src/ome2024_ngff_challenge/resave.py

```diff
@@ -856,16 +527,108 @@ def main(ns: argparse.Namespace, rocrate: ROCrateWriter | None = None) -> int:
     else:
         LOGGER.warning(f"no convertible metadata: {input_config.zr_attrs.keys()}")
 
+    if converted == 0:
+        raise SystemExit(1)
     return converted
 
 
-def cli(args=sys.argv[1:]):
+def cli(subparsers: argparse._SubParsersAction):
     """
     Parses the arguments contained in `args` and passes
     them to `main`. If no images are converted, raises
     SystemExit. Otherwise, return the number of images.
     """
-    parser = argparse.ArgumentParser()
+    cmd = "ome2024-ngff-challenge resave"
+    desc = f"""
+
+
+The `resave` subcommand will convert an existing Zarr v2 dataset into a Zarr v3 dataset according
+to the challenge specification. Additionally, a number of options are available for adding metadata
+
+
+
+BASIC
+
+    Simplest example:                        {cmd} --cc-by in.zarr out.zarr
+    Overwrite existing output:               {cmd} --cc-by in.zarr out.zarr --output-overwrite
+
+
+METADATA
+
+    There are three levels of metadata that the challenge is looking for:
+
+        - strongly recommended: license
+        - recommended: organism and modality
+        - optional: name and description
+
+    License: CC-BY (most suggested)          {cmd} in.zarr out.zarr --cc-by
+    License: public domain                   {cmd} in.zarr out.zarr --cc0
+    License: choose your own                 {cmd} in.zarr out.zarr --rocrate-license=https://creativecommons.org/licenses/by-sa/4.0/
+
+    Organism: Arabidopsis thaliana           {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid3702
+    Organism: Drosophila melanogaster        {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid7227
+    Organism: Escherichia coli               {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid562
+    Organism: Homo sapiens                   {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid9606
+    Organism: Mus musculus                   {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid10090
+    Organism: Saccharomyces cerevisiae       {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid4932
+
+    Modality: bright-field microscopy        {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000243
+    Modality: confocal microscopy            {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000251
+    Modality: two-photon laser scanning      {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000253
+    Modality: two-photon laser scanning      {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000253
```

Comment:
duplicated modality two-photon laser scanning. 
Do you want to put one for light-sheet microscopy instead?
`Modality: light-sheet microscopy (SPIM)    {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000369`

Reply:
Done. Thanks. (You guys can open PRs you know ;) )